### DISCUSSION TITLE
[Latest] Fork It And Run In Github Actions.....

### DISCUSSION THREAD

**Discussion by shivamguys** [shivamguys/irctc-cypress-automation#38] [Announcements]
If you are facing any issues running in your system Fork It , Update Passenger Data and run your code in Github Actions **Simple**.

Can Be Achieved In Less Than 5 Minute 
Note Running It For the **First Time**  would take some time as it would download everything and then would cache it from next time.
So be patient for the first time.

**If trying for tatkal run it prior 5 minutes tatkal time**

To Get Started Read this....  https://www.canva.com/design/DAGCmLdOM0g/5K21SNKKKZZMM4428K9wUw/view?utm_content=DAGCmLdOM0g&utm_campaign=designshare&utm_medium=link&utm_source=editor

**Comment by ujjwalbharadwaj** [comment]
This one also not worked for me, I have done all the steps as you guided.

**Reply by shivamguys** [reply]
I saw your cypress recording  you are booking for NDLS and the train destination is at ANVT and irctc shows a confirmation please use the exact source and destination...
see the recording of your error here @ujjwalbharadwaj 

![image](https://github.com/shivamguys/irctc-cypress-automation/assets/25263989/c4ebe26d-4eb5-45f0-9ea7-d51d2f60df6b)


https://cloud.cypress.io/projects/7afdkj/runs/102/test-results/8c153bd2-eb3b-4510-9c89-d39dc56c4819/replay?actions=%5B%5D&att=1&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D&ts=1713323136300.8013

**Reply by ujjwalbharadwaj** [reply]
@shivamguys Now workflow is running but I am not am able to see on screen that what is happening after running it, So can you guide about this?

**Reply by shivamguys** [reply]
can you tell me what was the error you were encountering on your system ? @ujjwalbharadwaj

**Reply by shivamguys** [reply]
to see the screen what's happening you have to run in your system .... @ujjwalbharadwaj

**Reply by ujjwalbharadwaj** [reply]
but in my system its giving an os error msg as i have told you earlier.... @shivamguys

**Reply by ujjwalbharadwaj** [reply]
![image](https://github.com/shivamguys/irctc-cypress-automation/assets/79165874/97c80f1a-3d1f-4e7b-b404-d7163a1438c8)
you can see in this image.

**Reply by ujjwalbharadwaj** [reply]
@shivamguys I have done setup of it and its running on my system but it is not entering user id and password

**Reply by shivamguys** [reply]
Are you on latest code?  @ujjwalbharadwaj

**Comment by ravi3517** [comment]
setup kiya hai aaj.. result kal dekhta hu

**Comment by shivamguys** [comment]
Are you on latest code ?

**Reply by ujjwalbharadwaj** [reply]
No I have cloned yesterday.

**Reply by ujjwalbharadwaj** [reply]
As you have guided in read me to enter username and password in env file so I have done that but it is not entering automatically.

**Reply by shivamguys** [reply]
Is username written in small or in upper case like USERNAME talking about the variable...

**Reply by ujjwalbharadwaj** [reply]
it was in small case i have changed that now its working......
Thank You for all your help😊

**Reply by oberoy-ravi** [reply]
Hi brother, am getting a runtime error. It is able to start but it unfortunately dues in 3rd test case. Kindly reply ASAP

**Reply by shivamguys** [reply]
I doubt you are on latest code because the latest code has all the fixes please compare your system code to the latest code present in mine repository 
Thanks
@ujjwalbharadwaj

**Reply by oberoy-ravi** [reply]
![error](https://github.com/shivamguys/irctc-cypress-automation/assets/114639806/2b374d09-513f-4dac-9466-4b2d41f7fe08)

GETTING THIS ERROR
@shivamguys

**Reply by oberoy-ravi** [reply]
Bhaiya @shivamguys , I have texted you on LinkedIn. Kindly provide your 10 mins.

-Thank you

**Reply by yas8say** [reply]
Bro how should we pay manually on workflow?

**Reply by shivamguys** [reply]
You should use your UPI ID in the workflow @yas8say

**Reply by shivamguys** [reply]
### Many of you might have been encountering this error when running workflow this is a very weird issue from cypress only I have tried to overcome this error you can try again and let me know if anything weird happens.....

Kindly **sync** to the latest code from the repo https://github.com/shivamguys/irctc-cypress-automation

```
We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: connect ECONNREFUSED 103.252.142.21:443

Common situations why this would fail:
  - you don't have internet access
  - you forgot to run / boot your web server
  - your web server isn't accessible
  - you have weird network configuration settings on your computer
      at <unknown> (http://localhost:37759/__cypress/runner/cypress_runner.js:131065:74)
      at visitFailedByErr (http://localhost:37759/__cypress/runner/cypress_runner.js:130463:12)
      at <unknown> (http://localhost:37759/__cypress/runner/cypress_runner.js:131064:11)
      at tryCatcher (http://localhost:37759/__cypress/runner/cypress_runner.js:1807:23)
      at Promise._settlePromiseFromHandler (http://localhost:37759/__cypress/runner/cypress_runner.js:1519:31)
      at Promise._settlePromise (http://localhost:37759/__cypress/runner/cypress_runner.js:1576:18)
      at Promise._settlePromise0 (http://localhost:37759/__cypress/runner/cypress_runner.js:1621:10)
      at Promise._settlePromises (http://localhost:37759/__cypress/runner/cypress_runner.js:1697:18)
      at _drainQueueStep (http://localhost:37759/__cypress/runner/cypress_runner.js:2407:12)
      at _drainQueue (http://localhost:37759/__cypress/runner/cypress_runner.js:2400:9)
      at Async._drainQueues (http://localhost:37759/__cypress/runner/cypress_runner.js:2416:5)
      at Async.drainQueues (http://localhost:37759/__cypress/runner/cypress_runner.js:22[86](https://github.com/n0steamash/irctc-cypress-automation/actions/runs/8770127211/job/24066365255#step:6:87):14)
  From Your Spec Code:
      at Context.eval (webpack://irctc-cypress-automation/./cypress/e2e/irctc.cy.js:24:7)
```

![Fork And Use It (1)](https://github.com/shivamguys/irctc-cypress-automation/assets/25263989/e6127c61-a317-432e-9324-e67cbc9f9bfa)


@ravi3517 @ujjwalbharadwaj @yas8say

**Reply by gunavardhan188** [reply]
why don't you make demo video😖?

**Reply by shivamguys** [reply]
This is not working it rarely work in GitHub actions try it on your machine

**Reply by princevish321** [reply]
I downloaded extension and directly trying to import it from extension menu but it show's manifest.json is missing so please give solution.

**Comment by Sapeksh-star** [comment]
i am getting this error:

python3: can't open file '/home/runner/work/irctc-cypress-automation/irctc-cypress-automation/irctc-captcha-solver/app.py': [Errno 2] No such file or directory
Error: Process completed with exit code 2.

**Comment by jeeiee** [comment]
Guys, please use ChatGPT or Cursor to solve issues.

**Comment by Shrey4390** [comment]
Where to set path in MAC 
I am getting below error


python3: can't open file '/home/runner/work/irctc-cypress-automation/irctc-cypress-automation/irctc-captcha-solver/app.py': [Errno 2] No such file or directory
Error: Process completed with exit code 2.

**Reply by itzavnish123** [reply]
go package.json and edit  this "python3 irctc-captcha-solver/app-server.py --host 0.0.0.0 --port 5000" and again run.......

**Reply by Shrey4390** [reply]
I have edited and commit changes but after that also I am getting below error. 

extensions-4.12.2
python3: can't open file '/home/runner/work/irctc-cypress-automation/irctc-cypress-automation/irctc-captcha-solver/app.py': [Errno 2] No such file or directory
Error: Process completed with exit code 2.


I think somewhere I need to change path. Please guide brother

**Comment by anujpal82** [comment]
Does it will retry if Tatkal quota is not opened yet ?
Because if quota it is not opened yet and again running all test cases definitely I will miss the ticket.

**Comment by Shrey4390** [comment]
Hi, can I have your number please.


8130080893

**Comment by anujpal82** [comment]
Hi @Shrey4390 
can we communicate through gmail ?
anujpal631998@gmail.com

**Comment by Shrey4390** [comment]
Hi,
If possible can you help me to setup in my system. Using Anydesk screen
sharing


8130080893

**Comment by Shrey4390** [comment]
Hi,
I am getting below error.
Please help

extensions-4.12.2
188
<https://github.com/Shrey4390/irctc-cypress-automation/actions/runs/13384562278/job/37378848903#step:4:189>python3:
can't open file
'/home/runner/work/irctc-cypress-automation/irctc-cypress-automation/irctc-captcha-solver/app.py':
[Errno 2] No such file or directory
189
<https://github.com/Shrey4390/irctc-cypress-automation/actions/runs/13384562278/job/37378848903#step:4:190>Error:
Process completed with exit code 2.

**Comment by mini7007** [comment]
hey @shivamguys    i am getting this error when i run the command provided in github 

1
clearCookies
2
clearLocalStorage
3
viewport1478, 1056
4
visithttps://www.irctc.co.in/nget/train-search
CypressError
cy.visit() failed trying to load:

https://www.irctc.co.in/nget/train-search

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

> Error: ESOCKETTIMEDOUT

Common situations why this would fail:
- you don't have internet access
- you forgot to run / boot your web server
- your web server isn't accessible
- you have weird network configuration settings on your computer
cypress/e2e/irctc.cy.js:24:8
  22 |     cy.clearLocalStorage()
  23 |     cy.viewport(1478, 1056)
> 24 |     cy.visit('https://www.irctc.co.in/nget/train-search', {
     |        ^
  25 |       failOnStatusCode: false,
  26 |       timeout: 90000
  27 |     })
  
  
  i tried to debug and found one solution that cypress is getting blocked from IRCTC every time it tries to connect then i make a playwright test code to check whether it is working or not that is accessing the IRCTC web but i want to solve above on cypress only can you please help me with this

**Reply by mini7007** [reply]
hey @shivamguys  or anyone can you help me with above

**Comment by stymrj** [comment]
@shivamguys!
Is this tool still working?

**Comment by lastbenchr** [comment]
@shivamguys I cloned code into my machine after completing setup and with further debugging, I was able to get access to irctc website and auto fill user name and password with auto correct captcha filled then after auto click on login, I was unable to login, it stays loading....trying to login attempt.  I figured out that the login api "webtoken" was kept pending... else give 504 Gateway Timeout (from service worker). I checked headers there is some differences in request headers, manual actual chrome browser vs  cypress.. after 7hours debugging I surrendered🥲.
![Screenshot 2025-04-28 091104](https://github.com/user-attachments/assets/25a68b79-0d9f-494b-967e-109f5b169aa4)

**Reply by mirtun786** [reply]
Any updates?

**Reply by shivamguys** [reply]
https://www.linkedin.com/posts/iamshivampandey_irctc-cypress-automationtesting-activity-7332728465625903104-H-nY?utm_source=share&utm_medium=member_ios&rcm=ACoAADD5k4oBdKU-xW1F6a-nDO3WfuDMxSVixQM

**Comment by itshop87** [comment]
manifest file missing error

**Comment by Pragathishbhagvan** [comment]
Hi @shivamguys ,

is this working now? I have seen your linkedin post. It would be great if you could provide an update on this.

**Comment by sirajshaikh-dev** [comment]
getting this error how to fix?
<img width="1920" height="1080" alt="Screenshot 2026-05-25 200503" src="https://github.com/user-attachments/assets/c403f899-32c6-4ed7-a582-6a2d01a3d0d6" />

**Comment by sirajshaikh-dev** [comment]
Irctc ESOCKETTIMEDOUT error, 
The IRCTC security is not allowing cypress to load irctc.co.in
